packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -868,24 +751,31 @@ function getCursorLeafNode(numberOfNodes: number, shape: TreeShape): UpPath {
 	}
 }
 
-function readEditableTree(tree: ISharedTree, numberOfNodes: number, shape: TreeShape) {
+function readEditableTree(tree: ISharedTree, numberOfNodes: number, shape: TreeShape): number {
+	let sum = 0;
 	switch (shape) {
 		case TreeShape.Deep: {
 			let currentNode: UnwrappedEditableField = tree.root;
-			for (let j = 0; j < numberOfNodes - 1; j++) {
+			for (let j = 0; j < numberOfNodes; j++) {
 				assert(isUnwrappedNode(currentNode));
+				const value = currentNode[valueSymbol] as number;
+				sum = applyOperationDuringRead(sum, value);
 				currentNode = currentNode.foo as UnwrappedEditableField;
 			}
-			break;
+			return sum;
 		}
 		case TreeShape.Wide: {
 			const root = tree.root;
 			assert(isUnwrappedNode(root));
 			const field = root.foo as UnwrappedEditableField;
 			assert(isEditableField(field));
-			for (const iterator of field) {
+			for (let i = 0; i < numberOfNodes; i++) {
+				const currentNode = field[i] as UnwrappedEditableField;
+				assert(isUnwrappedNode(currentNode));
+				const value = currentNode[valueSymbol] as number;
+				sum = applyOperationDuringRead(sum, value);
```

Comment:
if you tree ensures this is alwayse a number, you could just do

```suggestion
				sum += value as number;
```